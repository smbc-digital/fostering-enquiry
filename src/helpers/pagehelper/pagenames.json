[
	{
		"number": 1,
		"route": "/contact-the-fostering-team/about-yourself"
	},
	{
		"number": 2,
		"route": "/contact-the-fostering-team/contact-details"
	},
	{
		"number": 3,
		"route": "/contact-the-fostering-team/submit"
	}
	,
	{
		"number": 4,
		"route": "/contact-the-fostering-team/success"
	}
	,
	{
		"number": 5,
		"route": "/contact-the-fostering-team/error"
	}
]