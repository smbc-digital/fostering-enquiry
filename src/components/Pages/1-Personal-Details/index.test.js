import { React, mount} from '../../../helpers/SetupTest'
import { PersonalDetails } from './index'
import { getPageRoute } from '../../../helpers/pagehelper'
import renderer from 'react-test-renderer'

describe('PersonalDetails', () => {
    it('should call push on submit', () => {
        //Arrange
        const data = {
            firstName: {
				value:'Harry',
				isValid: true
			},
			lastName: {
				value:'Potter',
				isValid: true
			},
			dob: {
				value: '1980-07-31 00:00:00',
				isValid: true
            },
            onChange: jest.fn()
        }

        const history = { push: jest.fn() }
		const wrapper = mount(<PersonalDetails context={data} history={history} />)

        //Act
        wrapper.find('form').simulate('submit')


        //Assert
        expect(history.push).toHaveBeenCalledWith(getPageRoute(2))
    })

    describe('snapshot', () => {
		it('should render correctly',() => { 
			// Arrange
			const data = {
                firstName: {
                    value:'Harry',
                    isValid: true
                },
                lastName: {
                    value:'Potter',
                    isValid: true
                },
                dob: {
                    value: '1980-07-31 00:00:00',
                    isValid: true
                },
                onChange: jest.fn()
            }
            const history = {
                push: jest.fn()
            }
		
			// Act
			const tree = renderer
			.create(<PersonalDetails context={data} history={history} />)
			.toJSON()
		
			// Assert
			expect(tree).toMatchSnapshot()
		})
	})
})
