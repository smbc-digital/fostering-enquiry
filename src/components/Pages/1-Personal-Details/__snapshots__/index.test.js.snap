// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PersonalDetails snapshot should render correctly 1`] = `
<form
  onSubmit={[Function]}
>
  <h1 />
  <h2>
    Tell us about yourself
  </h2>
  <label
    className=""
    htmlFor="firstName"
  >
    First name
  </label>
  <input
    aria-label="First name"
    className=""
    id="firstName"
    inputMode={undefined}
    maxLength="35"
    min={undefined}
    name="firstName"
    onChange={[Function]}
    onKeyDown={[Function]}
    pattern={undefined}
    type="text"
    value="Harry"
  />
  <label
    className=""
    htmlFor="lastName"
  >
    Last name
  </label>
  <input
    aria-label="Last name"
    className=""
    id="lastName"
    inputMode={undefined}
    maxLength="60"
    min={undefined}
    name="lastName"
    onChange={[Function]}
    onKeyDown={[Function]}
    pattern={undefined}
    type="text"
    value="Potter"
  />
  <h3
    className="no-margin-bottom"
  >
    Date of birth
  </h3>
  <fieldset
    className="input-memorable-date"
  >
    <p>
      For example, 23 7 1987
    </p>
    <div>
      <label>
        Day
      </label>
      <input
        aria-label="day"
        className=""
        name="day"
        onChange={[Function]}
        onKeyDown={[Function]}
        type="number"
        value="31"
      />
    </div>
    <div>
      <label>
        Month
      </label>
      <input
        aria-label="month"
        className=""
        name="month"
        onChange={[Function]}
        onKeyDown={[Function]}
        type="number"
        value="7"
      />
    </div>
    <div>
      <label>
        Year
      </label>
      <input
        aria-label="year"
        className=""
        name="year"
        onChange={[Function]}
        onKeyDown={[Function]}
        type="number"
        value="1980"
      />
    </div>
  </fieldset>
  <button
    className="button-primary"
    disabled={false}
    id={undefined}
    type="submit"
  >
    Next step
  </button>
</form>
`;
